libsolidity/interface/CompilerStack.cpp

```diff
@@ -128,14 +129,13 @@ bool CompilerStack::parse()
 	vector<string> sourcesToParse;
 	for (auto const& s: m_sources)
 		sourcesToParse.push_back(s.first);
-	map<string, SourceUnit const*> sourceUnitsByName;
 	for (size_t i = 0; i < sourcesToParse.size(); ++i)
 	{
 		string const& path = sourcesToParse[i];
 		Source& source = m_sources[path];
 		source.scanner->reset();
 		source.ast = Parser(m_errors).parse(source.scanner);
-		sourceUnitsByName[path] = source.ast.get();
+		m_sourceUnitsByName[path] = source.ast.get();
```

Comment:
Actually, since you can also just use `m_sources[path].ast.get()` I think we should not add that member at all.